src/socket/cart.ts

```diff
@@ -0,0 +1,37 @@
+import {UsersProducts} from "../models/usersProducts";
+import {Product} from "../models/product";
+import {findProductByIdQuery} from "../db/db-queries";
+
+const usersProducts: UsersProducts = {};
+
+export const createUserCart = (userId: string) => {
+    usersProducts[userId] = {};
+};
+
+export const deleteUserCart = (userId: string) => {
+    usersProducts[userId] = {};
+};
+
+export const getUpdatedUserCart = async (socketId: string, productId: string): Promise<Partial<Product>> => {
+    const product: Product = await findProductByIdQuery(productId);
+    let updatedProductAmount: number = product.amount;
+    Object.keys(usersProducts).filter(userId => usersProducts[userId][productId])
```

Comment:
instead of checking if product exists and then reomve its amount you can write it shorter like:
updatedProductAmount -= usersProducts[usersId] || 0